### PR TITLE
update crd with upgrade fields

### PR DESCRIPTION
**Pull request by leena-jawale**
Signed-off-by: Leena Jawale <ljawale@redhat.com>

### CODE REVIEW

File: pkg/api/v1alpha1/clustercurator_types.go [hunk 1]
```diff
@@ -31,7 +31,7 @@ type ClusterCuratorSpec struct {
 	Destroy Hooks `json:"destroy,omitempty"`
 
 	// During an upgrade curation run these
-	Upgrade Hooks `json:"upgrade,omitempty"`
+	Upgrade UpgradeHooks `json:"upgrade,omitempty"`
```
Comment (leena-jawale) [inline pkg/api/v1alpha1/clustercurator_types.go:34]: Any suggestion for name instead of `UpgradeHooks` ?

**Comment by jnpacker** [comment]
@leena-jawale since you have the types.go open for editing, can you switch the ApiVersion to v1beta1?
https://github.com/open-cluster-management/backlog/issues/11252

**Comment by jnpacker** [comment]
Issue Ref: https://github.com/open-cluster-management/backlog/issues/11350

**Comment by leena-jawale** [comment]
> @leena-jawale since you have the types.go open for editing, can you switch the ApiVersion to v1beta1?
> [open-cluster-management/backlog#11252](https://github.com/open-cluster-management/backlog/issues/11252)

Yes will update version

File: pkg/api/v1beta1/clustercurator_types.go [hunk 2]
```diff
@@ -60,6 +60,28 @@ type Hooks struct {
 	OverrideJob *runtime.RawExtension `json:"overrideJob,omitempty"`
 }
 
+type UpgradeHooks struct {
+
+	// DesiredUpdate indicates the desired value of
+	// the cluster version. Setting this value will trigger an upgrade (if
+	// the current version does not match the desired version).
+	// +optional
+	DesiredUpdate string `json:"desiredUpdate,omitempty"`
```
Comment (hanqiuzh) [inline pkg/api/v1beta1/clustercurator_types.go:69]: update/upgrade? and it can be optional?

File: pkg/api/v1beta1/clustercurator_types.go [hunk 3]
```diff
@@ -60,6 +60,28 @@ type Hooks struct {
...
+
+	// Channel is an identifier for explicitly requesting that a non-default
+	// set of updates be applied to this cluster. The default channel will be
+	// contain stable updates that are appropriate for production clusters.
+	// +optional
+	Channel string `json:"channel,omitempty"`
```
Comment (hanqiuzh) [inline pkg/api/v1beta1/clustercurator_types.go:75]: where will hub users get the channel for the job? currently MCI only has available versions

and what is `non-default set of updates`? does that mean the field value should be default/non-default

File: pkg/api/v1beta1/clustercurator_types.go [hunk 4]
```diff
@@ -60,6 +60,28 @@ type Hooks struct {
...
+
+	// Upstream may be used to specify the preferred update server. By default
+	// it will use the appropriate update server for the cluster and region.
+	// +optional
+	Upstream string `json:"upstream,omitempty"`
```
Comment (hanqiuzh) [inline pkg/api/v1beta1/clustercurator_types.go:80]: not sure about the `appropriate update server` . is it always a fixed server for default?

File: pkg/api/v1beta1/clustercurator_types.go (hunk 2 above)
Comment (leena-jawale) [reply pkg/api/v1beta1/clustercurator_types.go:69]: yes agree it can't be optional

File: pkg/api/v1beta1/clustercurator_types.go (hunk 3 above)
Comment (leena-jawale) [reply pkg/api/v1beta1/clustercurator_types.go:75]: I copied this from clusterversions. I think default means the channel which is already in clusterversion resource

File: pkg/api/v1beta1/clustercurator_types.go (hunk 4 above)
Comment (jnpacker) [reply pkg/api/v1beta1/clustercurator_types.go:80]: I assume this text is just borrowed from the cluster version CRD.

**Pull request merged**